### PR TITLE
Feature/issue 17 - MoreScreen - Update UI

### LINKED ISSUE #17: [Feature] More Screen/ Update UI
### Overview
- // Update UI

### Impact
- // More Screen

### Screenshots (Optional)
- // Capture affected screen
Simulator Screen Shot - iPhone 12 Pro - 2021-10-22 at 15 11 53

### PR DESCRIPTION
**Pull request by minh5896**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // Update Ui

### Solution
- // Update UI

### Impact
- //  Screen

### Screenshots (Optional)
- // Capture affected screens
Simulator Screen Shot - iPhone 12 Pro - 2021-10-22 at 15 11 53

### CODE REVIEW

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 1]
```diff
@@ -7,11 +7,131 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+		guard let cell = self.downloadCollectionView.dequeueReusableCell(withReuseIdentifier: ConstantsCellDownload.downloadCell, for: indexPath) as? DownloadCell else { return UICollectionViewCell() }
+		cell.configure(with: viewModel?.downloadMovieViewModels[indexPath.row])
+		cell.layer.masksToBounds = true
+		cell.layer.cornerRadius = 5
+		cell.layer.borderWidth = 2
+		cell.layer.borderColor = UIColor.white.cgColor
+		return cell
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]:
```suggestion
```
@minh5896 : Move to Cell, not configure in View Controller

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 2]
```diff
@@ -7,11 +7,131 @@
...
+	func showError() {
+		downloadCollectionView.isHidden = true
+		view.addSubview(imageDownloadView())
+		view.addSubview(downloadButton())
+	}
+}
+// MARK: - UICollectionViewDelegateFlowLayout
+extension DownloadViewController: UICollectionViewDelegateFlowLayout {
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+		return CGSize(width: collectionView.frame.width / 3, height: collectionView.frame.height / 4 )
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: @minh5896 : Make constants, do not define magic number 3, 4 ...

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 3]
```diff
@@ -7,11 +7,131 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+		return 0
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: @minh5896 : Make constants, do not define magic number

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 4]
```diff
@@ -7,11 +7,131 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+		return 0
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: @minh5896 : Make constants, do not define magic number

File: MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift [hunk 5]
```diff
@@ -8,14 +8,32 @@
...
-	func getMovies() {
+	func getDownloadMovies() {
+		DependencyResolver.shared.movieAPIService.getDownloadMovies { [weak self] result in
+			guard let self = self else { return }
+			switch result {
+			case .success(let movies):
+				self.downloadMovieViewModels = movies.compactMap({
+					DownloadMovieViewModel(item: $0)
+				})
+//			self.viewController?.showError()
```
Comment (namnh92) [inline MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift]: @minh5896 : Remove comment

File: MovieApp/Screens/Home/HomeViewController.swift [hunk 6]
```diff
@@ -145,14 +148,39 @@ extension HomeViewController: UITableViewDataSource {
...
+		case .europe:
+			return "Europe"
+		case .romance:
+			return "Romance"
+		case .action:
+			return "Action"
+
+		}
+	}
+}
```
Comment (namnh92) [inline MovieApp/Screens/Home/HomeViewController.swift]: @
```suggestion

```
@minh5896 Move to Model

File: MovieApp/Screens/Download/ViewModels/ProfileTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		return profileData.count
+	}
+
+	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+		guard let cell = self.profileItem.dequeueReusableCell(withReuseIdentifier: ConstantsCellProfile.profileItemCell, for: indexPath) as? ProfileItemCell else { return UICollectionViewCell() }
+		cell.configure(with: profileData[indexPath.row])
+		cell.layer.masksToBounds = true
+		cell.layer.cornerRadius = 5
+		cell.layer.borderWidth = 2
+		cell.layer.borderColor = UIColor.white.cgColor
```
Comment (namnh92) [inline MovieApp/Screens/Download/ViewModels/ProfileTableViewCell.swift]:
```suggestion

```
@minh5896 Move to Cell, not configure in View Controller

**Review by namnh92 — CHANGES_REQUESTED** [review]

File: MovieApp/Screens/Home/HomeViewController.swift (hunk 6 above)
Comment (minh5896) [reply MovieApp/Screens/Home/HomeViewController.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 1 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 3 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift (hunk 5 above)
Comment (minh5896) [reply MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 4 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 2 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/ViewModels/StandardTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+class StandardTableViewCell: UITableViewCell {
+
+// MARK: - Iboutlets
+	@IBOutlet private weak var labelStandardCell: UILabel!
+
+// MARK: - Life Cycle
+	override func awakeFromNib() {
+		super.awakeFromNib()
+}
```
Comment (namnh92) [inline MovieApp/Screens/Download/ViewModels/StandardTableViewCell.swift:18]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/ce9aed5811734fd696fc2fb4d8ff634e8118eb8e/MovieApp/Screens/Download/ViewModels/StandardTableViewCell.swift#L18"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

File: MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift [hunk 9]
```diff
@@ -14,9 +14,9 @@ class TvShowViewCell: UITableViewCell {
 	@IBOutlet private weak var imageTvShow: UIImageView!
 	@IBOutlet private weak var playTvShowButton: UIButton!
 	@IBOutlet private weak var myListTvShowButton: UIButton!
+
 // MARK: - Life cycles
 	override func awakeFromNib() {
 		super.awakeFromNib()
-		// Initialization code
 	}
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/e147228c034589666db4398fcda9585015f8903c/MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift#L21"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

File: MovieApp/Screens/More/Views/MyListCell/MyListTableViewCell.swift [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+class MyListTableViewCell: UITableViewCell {
+
+// MARK: - Iboutlets
+	@IBOutlet private weak var imageMylist: UIImageView!
+	@IBOutlet private weak var myListlabel: UILabel!
+
+// MARK: - Life Cycle
+	override func awakeFromNib() {
+		super.awakeFromNib()
+}
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/MyListCell/MyListTableViewCell.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/e147228c034589666db4398fcda9585015f8903c/MovieApp/Screens/More/Views/MyListCell/MyListTableViewCell.swift#L19"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

**Comment by namnh92** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: MovieApp/Screens/Download/ViewModels/StandardTableViewCell.swift (hunk 8 above)
Comment (minh5896) [reply MovieApp/Screens/Download/ViewModels/StandardTableViewCell.swift:18]: fix complete

File: MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift (hunk 9 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift]: fix complete

File: MovieApp/Screens/More/Views/MyListCell/MyListTableViewCell.swift (hunk 10 above)
Comment (minh5896) [reply MovieApp/Screens/More/Views/MyListCell/MyListTableViewCell.swift]: fix complete

**Pull request closed without merging**